src/Mvc/test/Mvc.FunctionalTests/ComponentRenderingFunctionalTests.cs

```diff
@@ -17,18 +19,17 @@ public class ComponentRenderingFunctionalTests : IClassFixture<MvcTestFixture<Ba
         public ComponentRenderingFunctionalTests(MvcTestFixture<BasicWebSite.StartupWithoutEndpointRouting> fixture)
         {
             Factory = fixture;
-            Client = Client ?? CreateClient(fixture);
```

Comment:
The paradigm used in these tests was broken now that we report errors. The FetchData component is part of all of these tests, and when used with prerendering it was throwing an exception because the service configuration wasn't correct.

I can't help feeling like this is all more complicated than necessary. Using httpclient to call your own API isn't something we'd normally tell users to do.

Reply:
you can simply do `Client ?? CreateClient(fixture.WithWebHostBuilder(b => b.ConfigureServices(s => s.AddRazorComponents()))` or extend the MVC fixture and configure it there. Both work. I usually do the WithWebHostBuilder because I'm cheap and I don't bother with inheritance, but this is something you can do a nested class of the test and that's totally fine.